### PR TITLE
Roll forward for FX and additional paths

### PR DESCRIPTION
**Pull request by schellap**
@gkhanna79 PTAL.

---
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2360)

### CODE REVIEW

File: src/corehost/cli/fxr/fx_muxer.cpp [hunk 1]
```diff
@@ -45,28 +57,30 @@ pal::string_t fx_muxer_t::resolve_fx_dir(const pal::string_t& muxer_dir, runtime
...
+                most_compatible = std::max(ver, most_compatible);
             }
             if (specified.is_prerelease() && fx_ver_t::parse(version, &ver, false) && // false -- implies both production and prerelease.
                 ver.is_prerelease() && // prevent roll forward to production.
-                ver.get_major() == max_specified.get_major() &&
-                ver.get_minor() == max_specified.get_minor() &&
-                ver.get_patch() == max_specified.get_patch())
+                ver.get_major() == specified.get_major() &&
+                ver.get_minor() == specified.get_minor() &&
+                ver > specified)
```
Comment (schellap) [inline src/corehost/cli/fxr/fx_muxer.cpp]: @gkhanna79 do we need the patch to match for pre-releases?

Comment (gkhanna79) [inline src/corehost/cli/fxr/fx_muxer.cpp]: For pre-release, lets keep the experience consistent - rollforward within major.minor only.

**Comment by gkhanna79** [comment]
LGTM

File: src/corehost/cli/fxr/fx_muxer.cpp (hunk 1 above)
Comment (schellap) [inline src/corehost/cli/fxr/fx_muxer.cpp]: 1.0.0-rc2 rolls to 1.0.1-rc3?

Comment (schellap) [inline src/corehost/cli/fxr/fx_muxer.cpp]: Better, 1.0.0-rc2 rolls to 1.0.1-broken-alpha?

Comment (schellap) [inline src/corehost/cli/fxr/fx_muxer.cpp]: Cc @davidfowl

Comment (gkhanna79) [inline src/corehost/cli/fxr/fx_muxer.cpp]: Roll forwards can break even within 1.0.0-rc2 -> 1.0.0-rc2-a as well. If you want to keep it safe, check for patch too.

**Comment by schellap** [comment]
@dotnet-bot test this please.

**Comment by schellap** [comment]
@dotnet-bot test Ubuntu x64 Debug Build please
@dotnet-bot test Ubuntu x64 Release Build please
@dotnet-bot test Windows_NT x64 Release Build please

**Comment by schellap** [comment]
Fixes https://github.com/dotnet/cli/issues/2382

**Pull request merged**